apps/arweave/src/ar_disksup.erl

```diff
@@ -220,12 +220,28 @@ disk_free_cmd({unix, darwin}, Df, DataDirPath, Port) ->
 disk_free_cmd({unix, _}, Df, DataDirPath, Port) ->
 	 my_cmd(Df ++ " -Pa -B1 " ++ DataDirPath ++ "/", Port).
 
+%% check for hardware errors in df output
+check_for_hardware_error(DfOutput, ThrowOnError) ->
+    case lists:member("Input/output error", DfOutput) of
+        true ->
+            ar:console("~nERROR: one or more of your disks are in corrupt/failing state.~n~p~n", [DfOutput]),
+            if
```

Comment:
Because of how weirdly/confusingly erlang handles if statements (i.e `true` means `false` 🙃 ) The codebase usually uses case statement for everything. e.g.

```
case ThrowOnError of
    true ->
        erlang:error({input_output_error_detected, DfOutput});
    _ ->
        false
end
```

Reply:
fixed, true means, 1. it matches the error string we are on the lookout for and 2. true means HasError

Reply:
ah wait your comment is regarding the use of if and not case?

Reply:
it's now all case